clang/lib/Frontend/InitPreprocessor.cpp

```diff
@@ -1364,12 +1364,22 @@ static void InitializePredefinedMacros(const TargetInfo &TI,
   TI.getTargetDefines(LangOpts, Builder);
 }
 
+static void InitializePGOProfileMacros(const CodeGenOptions &CodeGenOpts,
```

Comment:
There are IR PGO users (e.g. Rust) which don't use the clang frontend. Do we need this macro based checks? 

I would prefer to just limit this patch to exposing the necessary interfaces. End-users can manage their own macros if they want to exclude the profile APIs from the source.

Reply:
This [comment](https://discourse.llvm.org/t/pgo-are-the-llvm-profile-functions-stable-c-apis-across-llvm-releases/75832/7?u=qwu_ibm) seems to indicate that such macros are useful, in addition to the reason that the compiler adding these macros can make the user program cleaner (user can now avoid wrapping these PGO calls or guarding them with macros). This PR does not change the current behaviour of `compiler-rt/lib/profile/InstrProfiling.h` so the user still has the freedom to use their own macros (or use the weak symbol mechanism).

Does this sound reasonable so we keep the macros? If not, I can split this into two PRs.

Reply:
Yes, my previous comment about other frontends was based on my (incorrect) understanding header inclusion in InstrProfiling.h. Adding the macros in this PR sgtm.

Reply:
Got it! Thanks for the clarification!